esp/bindings/http/platform/httpservice.cpp

```diff
@@ -974,3 +875,552 @@ int CEspHttpServer::onGet()
+}
+
+void CEspHttpServer::send200OK()
+{
+    m_response->setContentType("text/html; charset=UTF-8");
+    m_response->setStatus(HTTP_STATUS_OK);
+    m_response->send();
+}
+
+unsigned CEspHttpServer::createHTTPSession(EspAuthRequest& authReq, IPropertyTree* domainSessions, const char* _loginURL)
+{
+    CDateTime now;
+    now.setNow();
+    time_t createTime = now.getSimple();
+
+    StringBuffer peer;
+    VStringBuffer idStr("%s_%ld", m_request->getPeer(peer).str(), createTime);
+    unsigned sessionID = hashc((unsigned char *)idStr.str(), idStr.length(), 0);
+    ESPLOG(LogMax, "New sessionID <%d> at <%ld> in createHTTPSession()", sessionID, createTime);
+
+    Owned<IPropertyTree> ptree = createPTree();
+    ptree->addProp(PropSessionNetworkAddress, peer.str());
+    ptree->addPropInt64(PropSessionID, sessionID);
+    ptree->addPropInt(PropSessionState, HTTPSS_new);
+    ptree->setPropInt64(PropSessionCreateTime, createTime);
+    ptree->setPropInt64(PropSessionLastAccessed, createTime);
+
+    if (!isEmptyString(_loginURL))
+        ptree->addProp(PropSessionLoginURL, _loginURL);
+    else
+    {
+        StringBuffer loginURL = authReq.httpPath;
+        if (authReq.requestParams && authReq.requestParams->hasProp("__querystring"))
+            loginURL.append("?").append(authReq.requestParams->queryProp("__querystring"));
+        if (!loginURL.isEmpty() && streq(loginURL.str(), "/WsSMC/"))
+            ptree->addProp(PropSessionLoginURL, "/");
+        else
+            ptree->addProp(PropSessionLoginURL, loginURL.str());
+    }
+
+    if (domainSessions)
+    {
+        domainSessions->addPropTree(PathSessionSession, LINK(ptree));
+        return sessionID;
+    }
+
+    Owned<IRemoteConnection> conn = querySDSConnection(authReq.authBinding->queryDomainSessionSDSPath(), RTM_LOCK_WRITE, SESSION_SDS_LOCK_TIMEOUT);
+    conn->queryRoot()->addPropTree(PathSessionSession, ptree.getClear());
+    return sessionID;
+}
+
+IPropertyTree* CEspHttpServer::readAndCleanDomainSessions(EspHttpBinding* authBinding, IRemoteConnection* conn, time_t timeNow)
+{
+    if (!conn)
+        throw MakeStringException(-1, "Invalid SDS connection.");
+
+    if (authBinding->getSessionTimeoutSeconds() < 0)
+        return conn->queryRoot();
+
+    //Removing HTTPSessions if timed out
+    IArrayOf<IPropertyTree> toRemove;
+    IPropertyTree* root = conn->queryRoot();
+    Owned<IPropertyTreeIterator> iter = root->getElements(PathSessionSession);
+    ForEach(*iter)
+    {
+        IPropertyTree& item = iter->query();
+        if (timeNow - item.getPropInt64(PropSessionLastAccessed, 0) >= authBinding->getSessionTimeoutSeconds())
+            toRemove.append(*LINK(&item));
+    }
+    ForEachItemIn(i, toRemove)
+        root->removeTree(&toRemove.item(i));
+    return root;
+}
+
+bool CEspHttpServer::commitAndCloseRemoteConnection(IRemoteConnection* conn)
+{
+    if (!conn)
+        return false;
+
+    conn->commit();
+    conn->close();
+    return true;
```

Comment:
Is commitAndCloseRemoteConnection() method ever necessary? (most cases I can see, it is not)

Code like : 
```
block()
{
   Owned<IRemoteConnection> conn = querySDS().connect(.....)
    IPropertyTree *root = conn->queryRoot();
    <add stuff to root>
}
```

doesn't need to call conn->commit() or conn->close() they are implicit when IRemoteConnection is destroyed (when it goes out of scope)